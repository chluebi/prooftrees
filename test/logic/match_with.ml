open Logic.Logic_exp.Examples

let () =
  let inputs =
    [
      (tree1, treeA, tree3);
      (tree1, tree3, tree1);
      (treeA, treeB, treeB);
      (tree4, treeA, treeA);
      (tree4, treeA, tree3);
      (tree1, tree1, tree1);
    ]
  in
  let f (tree, structure, ass_tree) =
    match T.match_with tree structure with
    | Some ass -> print_endline (T.to_string (T.assign ass ass_tree))
    | None -> print_endline "No assignment"
  in
  List.iter f inputs
