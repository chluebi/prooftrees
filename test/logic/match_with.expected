Or(Or(A, B), Or(A, B))
No assignment
A
Not(A)
Or(Not(A), Not(A))
Or(A, B)
